server/src/main/java/org/apache/druid/metadata/MetadataSegmentManager.java

```diff
@@ -59,6 +60,8 @@
 
   Collection<ImmutableDruidDataSource> getDataSources();
 
+  Iterable<DataSegment> iterateAllSegments();
```

Comment:
I intentionally renamed the method to not start with "get". Although formally it returns a lambda whose creation is cheap, iteration itself involves `flatMap()`, creation of many streams, and therefore is not so cheap. 

See item 16 in [checklist](https://docs.google.com/document/d/17EEKT6fih9Dd5NfXjBoECcKbVp1eOB2vb3jKqTF9pPc/edit).

I will add Javadoc to this method to additionally highlight that the iteration is not super cheap (however, far cheaper than materialization to collections that was done before.)

Reply:
Appears that I (without realizing) used the same prefix as in `SegmentId.iterateAllPossibleParsings()` (which is also a "lazy and not cheap iteration"). Looks like an emerging naming pattern.

Reply:
I also feel this would iterate itself rather than returning an iterable. Even if the iteration can be not cheap, I think it should be fine to call `getIterable` because returning an iterable is cheap.

Reply:
Creation of the iterable is cheap, but the creation of the underlying *iteration* is not cheap. This method discourages iterating twice on the result, and, in fact, even storing the result in a variable, but rather immediately consuming it, as it happens in all use cases: they look either like
```java
for (DataSegment s : iterateAllSegments()) {
  ..
}
```
or
```java
iterateAllSegments().forEach(s -> ...);
```
or feeded into `Iterables.transform()`, that happens in `DruidCoordinatorSegmentInfoLoader`.

I think the prefix `iterate` reflects this sense of discouragement well. People may feel a little uncomfortable storing an iterable returned from such method in a variable, but the constructions above look perfectly OK. `getIterable()` doesn't such sense of uncomfortableness.

Reply:
Ok, my main complaint was less on using 'get' and more on 'iterate' over 'iterable', but I think maybe this is ok based on your explanation. I would likely check out the javadoc if i encountered this method, so I think it checks out? :+1:

Reply:
Yes, there is an intention to provoke readers to go and see the Javadoc when they encounter a method with such a name.